File: solutions/0007-Reverse-Integer.js
/**
 * @param {number} x
 * @return {number}
 */
const reverse = function(x) {
  const result = Math.abs(x).toString().split('').reverse().join('');

  if (result > Math.pow(2, 31)) return 0;
  if (x >= 0) return parseInt(result);
  
  return parseInt(result * -1);
};

File: solutions/0007-Reverse-Integer.cpp
class Solution {
public:
	int reverse(int x) {
		auto pos = std::abs(x);
		auto s_num = std::to_string(pos);
		std::reverse(s_num.begin(), s_num.end());
		auto ri = std::stol(s_num);
		if (ri < INT_MIN || ri > INT_MAX) return 0;
		return x < 0 ? ri * -1 : ri; 
  }
};
